### PR TITLE
#3203 add table to show currently standing offers for user

### LINKED ISSUE #3203: add page to show currently standing offers for user
currently, there's no way to see all of the offers users made, would be nice to have this in some sort of table. 
- add allow for easier withdrawal of offer
- test if it's showing offers on burned nfts where offers did not expire

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3203
- [ ] Requires deployment <rubick/magick_issue>
- [x] Added another tab in users profile section named 'Offers made'

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-07-18 at 9 15 21 PM

### CODE REVIEW

**Comment by yangwao** [comment]
I would sort the newest offers on top by default.

Can we make columns sortable for amount, date and name? 

image

File: components/rmrk/Profile/ProfileDetail.vue [hunk 1]
```diff
@@ -836,6 +876,9 @@ export default class Profile extends mixins(
     if (this.activeTab === 'sales') {
       this.fetchSalesEventByCreator()
     }
+    if (this.activeTab === 'offers') {
+      this.fetchOfferEvents()
+    }
```
Comment (Jarsen136) [inline components/rmrk/Profile/ProfileDetail.vue:881]: let's try to make a function to reuse this stuff

      if (this.activeTab === 'history') {
        this.fetchCollectionEvents()
      }
      if (this.activeTab === 'sales') {
        this.fetchSalesEventByCreator()
      }
      if (this.activeTab === 'offers') {
        this.fetchOfferEvents()
      }

File: queries/subsquid/bsx/offerListUser.graphql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+query offerListUser($id: String!) {
+  offerEvents(where: { caller_eq: $id }) {
+    offer {
```
Comment (Jarsen136) [inline queries/subsquid/bsx/offerListUser.graphql:3]: Seem we do not need `offerEvents`.
There is a data named  `createdAt` in `offer` query.

**Comment by petersopko** [comment]
> Can we make columns sortable for amount, date and name?

Can we please make the first column shortened? Perhaps add ... if the name of the nft is too long? Table seems to not be centered.

**Review by petersopko — CHANGES_REQUESTED** [review]
left the comment about first column, I guess I meant it in the similar fashion to https://beta.kodadot.xyz/bsx/stats

**Comment by prachi00** [comment]
> > Can we make columns sortable for amount, date and name?
> 
> Can we please make the first column shortened? Perhaps add ... if the name of the nft is too long? Table seems to not be centered.

done

**Comment by prachi00** [comment]
> seems works https://deploy-preview-3477--koda-nuxt.netlify.app/bsx/u/bXgvP2cQzEnPvuPQHRghz9dUYovqzyPhXBRXbNwVCef7nsmqb?tab=offers&page=1
> 
> I would sort the newest offers on top by default.
> 
> Can we make columns sortable for amount, date and name?
> 
> image

@yangwao done

**Comment by petersopko** [comment]
wfm, just one thing, did we leave out the sorting by offer status on purpose? Maybe it'd be nice to have option to show only active offers since those are the ones which will be most interesting to user? @yangwao wdyt

**Comment by yangwao** [comment]
> did we leave out the sorting by offer status on purpose?

seems works on https://beta.kodadot.xyz/bsx/stats/ so let's have it there @prachi00 

image

> Maybe it'd be nice to have option to show only active offers since those are the ones which will be most interesting to user?

legit, add it here 
- #3463

**Review by petersopko — APPROVED** [review]
wfm

**Review by Jarsen136 — APPROVED** [review]
LGTM

File: queries/subsquid/bsx/offerListUser.graphql [hunk 3]
```diff
@@ -0,0 +1,15 @@
+query offerListUser($id: String!) {
+  offers(orderBy: createdAt_DESC, where: { caller_eq: $id }) {
+    blockNumber
+    caller
+    createdAt
+    id
+    price
+    status
+    updatedAt
+    nft {
```
Comment (Jarsen136) [inline queries/subsquid/bsx/offerListUser.graphql:10]: it should limit for burned nft

**Review by Jarsen136 — COMMENTED** [review]
I bet ths query does not meet this point. 👀 

image

File: queries/subsquid/bsx/offerListUser.graphql (hunk 3 above)
Comment (prachi00) [reply queries/subsquid/bsx/offerListUser.graphql:10]: but doesnt it mean that it should show up on burned nfts?

Comment (Jarsen136) [reply queries/subsquid/bsx/offerListUser.graphql:10]: cc @petersopko

Comment (petersopko) [reply queries/subsquid/bsx/offerListUser.graphql:10]: we want to show burned NFTs so user can withdraw his offers

**Review by petersopko — APPROVED** [review]

**Comment by petersopko** [comment]
@yangwao I see that you've added this reuqest
> test if it's showing offers on burned nfts where offers did not expire

did you mean to add some column which will track whether NFT is burned or not? Cause the query right now, is fetching all of the NFTs regardless of their state (so you can see the burned ones in the table)

**Comment by yangwao** [comment]
> did you mean to add some column which will track whether NFT is burned or not?

Let's make a separate issue 

> Cause the query right now, is fetching all of the NFTs regardless of their state (so you can see the burned ones in the table)

✅

**Comment by prachi00** [comment]
> > did you mean to add some column which will track whether NFT is burned or not?
> 
> Let's make a separate issue
> 
> > Cause the query right now, is fetching all of the NFTs regardless of their state (so you can see the burned ones in the table)
> 
> ✅

yeah I can make that but kinda need more clarity on this one

**Comment by prachi00** [comment]
pay 250 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $250 @ 68.3 USD/KSM ~ 3.66 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0xe2b576022bf77da2d6934e5e464a1821a48211db7c408ac6e1ee4b134b806902](https://kusama.subscan.io/extrinsic/0xe2b576022bf77da2d6934e5e464a1821a48211db7c408ac6e1ee4b134b806902)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**